Backends/CLX/clipboard.lisp

```diff
@@ -0,0 +1,227 @@
+                                                  :encoding :utf-8)))
+             (process-string-reply port selection content :string)))
+          ((:|text/html|)
+           (let ((content (babel:octets-to-string (coerce (xlib:get-property window property) '(vector (unsigned-byte 8)))
+                                                  :encoding :utf-8)))
+             (process-string-reply port selection content :html)))
+          ((:|image/png|)
+           (let ((image (convert-image-array (coerce (xlib:get-property window property) '(vector (unsigned-byte 8))) :png)))
+             (process-string-reply port selection image :image)))
+          ((:|image/jpeg|)
+           (let ((image (convert-image-array (coerce (xlib:get-property window property) '(vector (unsigned-byte 8))) :jpeg)))
+             (process-string-reply port selection image :image)))
+          (t
+           nil))))))
+
+(defun process-selection-clear (port selection)
+  (set-stored-object port selection nil))
+
+;;;
+;;;  Paste support
+;;;
+
+(defmethod clx-request-clipboard-content ((port clx-clipboard-port-mixin) pane selection type)
+  (setf (clipboard-outstanding-request-pane port) pane)
+  (setf (clipboard-outstanding-request-type port) (if (listp type) type (list type)))
+  (setf (clipboard-outstanding-request-selection port) selection)
+  (xlib:convert-selection selection :targets (sheet-direct-xmirror pane) :mcclim nil))
+
+(defmethod climb:request-selection-content ((port clx-clipboard-port-mixin) pane type)
+  (clx-request-clipboard-content port pane :primary type))
+
+(defmethod climb:request-clipboard-content ((port clx-clipboard-port-mixin) pane type)
+  (clx-request-clipboard-content port pane :clipboard type))
+
+(defun process-targets-reply (port window selection time)
+  (when (and (clipboard-outstanding-request-pane port)
+             (eq (clipboard-outstanding-request-selection port) selection))
+    (let* ((display (xlib:window-display window))
+           (targets (mapcar (lambda (v)
+                              (xlib:atom-name display v))
+                            (xlib:get-property window :mcclim)))
+           (selected-type (loop
+                            named outer
+                            for request-type in (clipboard-outstanding-request-type port)
+                            for v = (loop
+                                      for type in (representation-type-to-native request-type)
+                                      when (member type targets)
+                                        return type)
+                            when v
+                              return v)))
+      (if selected-type
+          (xlib:convert-selection selection selected-type window :mcclim time)
+          ;; ELSE: The clipboard doesn't support content of this type, send a negative response here
+          nil))))
+
+(defun process-string-reply (port selection content type)
```

Comment:
this function name is bogus, reply may be image according to above. `process-clipboard-reply`?

Reply:
I agree. Changed.